packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -114,17 +114,14 @@ class Observatory {
     });
   }
 
-  Future<Response> reloadSources(String isolateId) async {
-    Completer<Event> whenIsolateReloads = new Completer<Event>();
-    StreamSubscription<Event> sub = onIsolateEvent
-      .where((Event event) => event.kind == 'IsolateReload')
-      .listen((Event event) => whenIsolateReloads.complete(event));
-
+  // Returns null on success and the reload failure reason on failure.
+  Future<String> reloadSources(String isolateId) async {
```

Comment:
The reload RPC is synchronous and doesn't return until the reload has succeed or failed.